Cython/Compiler/ModuleNode.py

```diff
@@ -358,7 +358,9 @@ def generate_c_code(self, env, options, result):
 
         code = globalstate['before_global_var']
         code.putln('#define __Pyx_MODULE_NAME "%s"' % self.full_module_name)
-        code.putln("int %s%s = 0;" % (Naming.module_is_main, self.full_module_name.replace('.', '__')))
+        module_is_main = "%s%s" % (Naming.module_is_main, self.full_module_name.replace('.', '__'))
+        code.putln("extern int %s;" % module_is_main)
+        code.putln("int %s = 0;" % module_is_main)
```

Comment:
I don't understand this. Why is it both `extern` and internal? And why does it need to be `extern` at all? Shouldn't it be `static` instead?

Reply:
@scoder Maybe is not really required, but look at `bin/cython_freeze`. Are you 100% sure that making it `static` will not break things?

Reply:
@scoder I can definitely confirm that `static` breaks `Demos/freeze`

```diff
[dalcinl@kl-13999 cython]$ git diff
diff --git a/Cython/Compiler/ModuleNode.py b/Cython/Compiler/ModuleNode.py
index 3c04f0838..e559cd43a 100644
--- a/Cython/Compiler/ModuleNode.py
+++ b/Cython/Compiler/ModuleNode.py
@@ -358,7 +358,7 @@ class ModuleNode(Nodes.Node, Nodes.BlockNode):
 
         code = globalstate['before_global_var']
         code.putln('#define __Pyx_MODULE_NAME "%s"' % self.full_module_name)
-        code.putln("int %s%s = 0;" % (Naming.module_is_main, self.full_module_name.replace('.', '__')))
+        code.putln("static int %s%s = 0;" % (Naming.module_is_main, self.full_module_name.replace('.', '__')))
         code.putln("")
         code.putln("/* Implementation of '%s' */" % env.qualified_name)
```

```
[dalcinl@kl-13999 cython]$ PYTHONPATH=$PWD make -C Demos/freeze
../../bin/cython_freeze combinatorics lcmath > nCr.c
gcc -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra -I/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7  -c -o nCr.o nCr.c
../../bin/cython  combinatorics.pyx
gcc -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra -I/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7  -c -o combinatorics.o combinatorics.c
../../bin/cython  lcmath.pyx
gcc -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra -I/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7  -c -o lcmath.o lcmath.c
gcc -u _PyMac_Error /usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/Python -L/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config  nCr.o combinatorics.o lcmath.o  -L/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/lib -L/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config -lpython2.7 -o nCr
Undefined symbols for architecture x86_64:
  "___pyx_module_is_main_combinatorics", referenced from:
      _main in nCr.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [nCr] Error 1
```